internal/resources/targetlocation/resource_target_location.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	diags = resourceTargetLocationRead(ctx, data, m)
+
+	return diags
+}
+
+func resourceTargetLocationImporter(_ context.Context, data *schema.ResourceData, config any) ([]*schema.ResourceData, error) {
+	client := config.(authctx.TanzuContext)
+
+	targetLocationName := data.Id()
+
+	if targetLocationName == "" {
+		return nil, errors.New("ID is needed to import an TMC AKS cluster")
+	}
+
+	namesArray := strings.Split(targetLocationName, "/")
+
+	if len(namesArray) != 2 {
+		return nil, errors.Errorf("Invalid target location ID.\nTarget location id should consists of a provider namer and the target location name separated by '/'.\nProvided ID: %s", targetLocationName)
+	}
+
+	targetLocationFn := &targetlocationmodels.VmwareTanzuManageV1alpha1DataprotectionProviderBackuplocationFullName{
+		ProviderName: namesArray[0],
+		Name:         namesArray[1],
+	}
+	getResponse, err := client.TMCConnection.TargetLocationService.TargetLocationResourceServiceGet(targetLocationFn)
+
+	if err != nil || getResponse.BackupLocation == nil {
+		return nil, errors.Errorf("Couldn't Import backup target location. Name: %s, Provider: %s", targetLocationFn.Name, targetLocationFn.ProviderName)
+	} else {
+		credentialsType, err := getCredentialsType(client, getResponse.BackupLocation.Spec.Credential.Name)
+
+		if err != nil {
+			return nil, errors.Errorf("Couldn't Import backup target location. Credentials Error: %s", err.Error())
+		}
+
+		modifyResponse(getResponse.BackupLocation, credentialsType)
+		err = tfModelResourceConverter.FillTFSchema(getResponse.BackupLocation, data)
+
+		if err != nil {
+			return nil, err
+		}
+
+		data.SetId(targetLocationFn.Name)
+	}
+
+	return []*schema.ResourceData{data}, err
+}
+
+func validateSchema(tlModel *targetlocationmodels.VmwareTanzuManageV1alpha1DataprotectionProviderBackuplocationBackupLocation, credentialsType credentialsmodels.VmwareTanzuManageV1alpha1AccountCredentialProvider) error {
+	if credentialsType == credentialsmodels.VmwareTanzuManageV1alpha1AccountCredentialProviderAWSEC2 {
+		if tlModel.Spec.Bucket != "" || tlModel.Spec.Config != nil {
+			return errors.Errorf("Bucket field and Config block should not be set when credentials are %s", credentialsType)
+		}
+
+		tlModel.Spec.Config = &targetlocationmodels.VmwareTanzuManageV1alpha1DataprotectionProviderBackuplocationTargetProviderSpecificConfig{}
+	} else {
+		if tlModel.Spec.Bucket == "" || tlModel.Spec.Config == nil {
+			return errors.Errorf("Bucket field and Config block must be set when credentials are %s", credentialsType)
+		}
+
+		if tlModel.Spec.Config.S3Config != nil && tlModel.Spec.Config.AzureConfig != nil {
+			return errors.New("Config should be set with either AWS or Azure blocks but not both.")
+		}
+	}
+
+	return nil
+}
+
+func modifyResponse(tlModel *targetlocationmodels.VmwareTanzuManageV1alpha1DataprotectionProviderBackuplocationBackupLocation, credentialsType credentialsmodels.VmwareTanzuManageV1alpha1AccountCredentialProvider) {
```

Comment:
nit: maybe add a comment explaining why this is needed?

Reply:
Added a comment